doc/development/tutorials/overview.rst

```diff
@@ -0,0 +1,32 @@
+Developing extensions overview
```

Comment:
This document is still in the wrong place, IMO. Could we move it up a level? As with the builders document, this is general reference material, not tutorial-style content

Reply:
Just for reference, my long running doc rework is based on how [Django do their documentation](https://jacobian.org/2009/nov/10/what-to-write/) (that's a blog from one of the founders of the project, iirc). This document falls into the topical guide category, IMO

Reply:
just did this in the latest commit 👍